### ISSUE TITLE
not working in  magento 1.9.1.0

### ISSUE THREAD

**Issue by avijit12** [organicinternet/magento-configurable-simple#224]
1)Getting sql error
2)Indexing (price) not working

please waiting for solution

Avijit Nanda

**Comment by Devrons** [comment]
having the same issue.

**Comment by insit** [comment]
I just fixed the Indexing product prices issue on my environment, latest 1.9.1
Apparently there is a difference between the Configurable.php class of the package installed via Magento Connect and this one http://pastebin.com/index/spVj88T4. Compared it via GIT and this one adds two columns called group_price and base_group_price to the $select and $outerSelect objects.

So use this class io the one in the package and the re-indexing should work. Fixes displaying wrong prices in products overview pages. For the rest awesome extension, can't understand why Magento does not offer this functionality by default, XCart does.

**Comment by Devrons** [comment]
Not to be a dunce, but could you tell me where Configurable.php is located. I have run a search and browsed through all of the folders to no avail. Thank you for your help!

**Comment by insit** [comment]
/app https://github.com/insit/magento-configurable-simple/tree/master/app/
code
https://github.com/insit/magento-configurable-simple/tree/master/app/code/
community
https://github.com/insit/magento-configurable-simple/tree/master/app/code/community
/OrganicInternet
https://github.com/insit/magento-configurable-simple/tree/master/app/code/community/OrganicInternet
/SimpleConfigurableProducts
https://github.com/insit/magento-configurable-simple/tree/master/app/code/community/OrganicInternet/SimpleConfigurableProducts
/CatalogInventory
https://github.com/insit/magento-configurable-simple/tree/master/app/code/community/OrganicInternet/SimpleConfigurableProducts/CatalogInventory
/Model
https://github.com/insit/magento-configurable-simple/tree/master/app/code/community/OrganicInternet/SimpleConfigurableProducts/CatalogInventory/Model
/Mysql4
https://github.com/insit/magento-configurable-simple/tree/master/app/code/community/OrganicInternet/SimpleConfigurableProducts/CatalogInventory/Model/Mysql4
/Indexer
https://github.com/insit/magento-configurable-simple/tree/master/app/code/community/OrganicInternet/SimpleConfigurableProducts/CatalogInventory/Model/Mysql4/Indexer
/Stock
https://github.com/insit/magento-configurable-simple/tree/master/app/code/community/OrganicInternet/SimpleConfigurableProducts/CatalogInventory/Model/Mysql4/Indexer/Stock
/_Configurable.php_

_You can try to get it from
here: https://github.com/insit/magento-configurable-simple/blob/master/app/code/community/OrganicInternet/SimpleConfigurableProducts/CatalogInventory/Model/Mysql4/Indexer/Stock/Configurable.php
https://github.com/insit/magento-configurable-simple/blob/master/app/code/community/OrganicInternet/SimpleConfigurableProducts/CatalogInventory/Model/Mysql4/Indexer/Stock/Configurable.php_

2015-02-09 2:47 GMT+01:00 Devrons notifications@github.com:

**Comment by alexcurtis** [comment]
@insit 
If I replace the Configurable.php with your version and run an index through the shell. I get an error:

PHP Fatal error:  Call to a member function setTypeId() on a non-object in ~/webstore/app/code/core/Mage/CatalogInventory/Model/Resource/Indexer/Stock.php on line 278

Many Thanks.

**Comment by phele** [comment]
Hi,

I followed your instruction to replace Configurable.php file with the updated file, but I get this error when indexing Product Prices.

Fatal error: Call to a member function setTypeId() on a non-object in /home/advertising/public_html/domainname.com/app/code/core/Mage/CatalogInventory/Model/Resource/Indexer/Stock.php on line 278

I'm using Magento 1.9.1.0. Please help!

Thanks

**Comment by CasperMunk** [comment]
I get the same error as phele. Does anybody have an idea how to fix this?

EDIT: Seems like this fork is updated and works with 1.9.1.0

**Comment by phele** [comment]
Wow. After unsuccessfully trying this fix on version 1.9.1.0, 1.9.0.1 and 1.8.10, I read the reviews from http://www.magentocommerce.com/magento-connect/simple-configurable-products.html and found out that the path to replace Configurable.php should be app/code/community/OrganicInternet/SimpleConfigurableProducts/Catalog/Model/Resource/Eav/Mysql4/Product/Indexer/Price/

It's working for me on those version now. Good luck, everyone!

**Comment by kevinm2** [comment]
Should the latest package be downloaded from MagentoConnect or here (Github)?  If the latest should be downloaded from github, can I just copy the file structure to my magento folder?  Or it has to be done through MagentoConnect?

**Comment by SheriSmith** [comment]
Like Phele said, you need to create the folder Price/ in app/code/community/OrganicInternet/SimpleConfigurableProducts/Catalog/Model/Resource/Eav/Mysql4/Product/Indexer/

Then you need to add the Configurable.php file to the newly made Price/ folder.  

Don't use Magento Connect.  Download the zip folder from Github.  Don't forget to clear the cache, log out and log in for good measure.  

This works in CE 1.9.  Yay!

**Comment by mrcodexcy** [comment]
As everyone mentioned above, if you install the SCP through magento connect you might end up with a broken installation which I believe the problem lies when Magento extracts the files the length of that directory and file are way too long (more than 260 chars) thus the problem.

**Comment by jasmeetdd** [comment]
@phele and @kevinm2 and @SheriSmith and @mrcodexcy 
Thanks buddy you saved me a life. I was asking my client to buy this extension:-
https://www.magentocommerce.com/magento-connect/configurable-products-use-simple-price-and-product-details.html
 as I thought SCP was not compatible with 1.9.0. He was not ready to buy that extension. But you made my life. I have downloaded the latest version from GitHub. Copied the new configurable.php file from:-
http://pastebin.com/spVj88T4
To:-
app/code/community/OrganicInternet/SimpleConfigurableProducts/Catalog/Model/Resource/Eav/Mysql4/Product/Indexer/Price/
And viola! Life Saved.

Thanks @kevinm2 for asking that very important question and thanks @SheriSmith & @mrcodexcy for answering it.

Thank You!

**Comment by npire1** [comment]
I have tried the above i.e have replaced the file with the one from pastebin and It works as before except when you pick the simple product from the drop down list, suddenly the screen duplicates itself and creates another copy to the right. Only way to stop It is to put the product page as zero columns. scp Is affecting my theme layout ?

**Comment by SheriSmith** [comment]
npire1,

Are you clearing out all of your cache? Magento cache and any other cache like memcache, apc, cloudflare, etc? Logging out and in again?

Did you download the zip file directly from Github and not Magento Connect?

If yes to all of the above, start from scratch and do it all over again, it's possible that in the downloading and unzipping of everything that something got messed up. 

If it is still messed up, then switch to the default Magento theme. If after switching themes you see that it works, then you know your problems have to do with your theme.

**Comment by npire1** [comment]
Hi Thanks

Well I have It all loaded via magento connect and then I have "Copied the new configurable.php file from:-
http://pastebin.com/spVj88T4
To:-
app/code/community/OrganicInternet/SimpleConfigurableProducts/Catalog/Model/Resource/Eav/Mysql4/Product/Indexer/Price/


So should I uninstall the magento connect version then download a version from Github ? If so I cant find the Github version ? 

Thanks again

**Comment by npire1** [comment]
I have cleared the cache and reindexed etc but I am not sure how to clear memcache, apc, cloudflare, etc

Yes If I uninstall scp then the issue goes away but changing theme does not help.

**Comment by SheriSmith** [comment]
Yes, you need to uninstall the Magento Connect Version. Be sure to clear out all cache and reindex after you uninstalled it, and log out and in for good measure.

Go to:
https://github.com/organicinternet/magento-configurable-simple

Click on the green download button. You will need to have FTP and/or SSH shell access to your server to handle this zipped file. Then you can add the Price/ folder and copy the configurable.php file to that new folder. Be sure the folder has 0755 permissions.

Clear your Magento Cache, reindex and log out and in again. This should fix everything. 

You will have to see if you have any other kinds of cache associated with your server and/or website. If you don't know of any, then you probably don't have anything beyond the cache that is built into Magento.

**Comment by npire1** [comment]
Oh great, I get you ok I will give It a go...

Many thanks, big help :)

**Comment by npire1** [comment]
That's fixed !!!

Thank you so much for all the help :)

**Comment by fairport** [comment]
Hello..
I have just upgraded from 1.9.2.1 to 1.9.3.4 and I am getting the exact same error message as @phele 
PHP Fatal error:  Call to a member function setTypeId() on boolean in ..../public_html/app/code/core/Mage/CatalogInventory/Model/Resource/Indexer/Stock.php on line 278

Reindexing throws a 500 error and the error_log file gets the error message above.

 I have "Copied the new configurable.php file from:-
http://pastebin.com/spVj88T4
To:-
app/code/community/OrganicInternet/SimpleConfigurableProducts/Catalog/Model/Resource/Eav/Mysql4/Product/Indexer/Price/

Not fixed. Any new ideas on this issue? Has anyone else upgraded to 1.9.3.4 w/o issues ?

Thanks!

**Comment by jasmeetdd** [comment]
Well if you are going to use Magento 1.9.x , then I would suggest you to stick to Magento 1.9.0 only. It is compatible with most of the major plugin.

Whenever I upgraded I got serious problems. You can still install critical security updates though.

**Comment by fairport** [comment]
That is not very helpful at all. I have been on 1.9.2.1 for several years.

The last "critical" patch  I installed broke the images browse & upload buttons away. It introduced some 1.9.3.4 functionality & in that version, they no longer use flash for uploading images.

Any other sugestions ?
